refactor(vscode): improve statusbar item display

Sysix opened the pull request:
This PR refactors the VSCode extension's status bar item display logic .
- Double Checkmark: Both tools (oxlint / oxfmt) are found and can be used
- Single Checkmark: One of the tools binary could not be found and is deactivated
- Circle: None of tools binary could be found

The background color changed a bit too. The error background is not shown anymore, only the warning background when no binaries could be found.

closes https://github.com/oxc-project/oxc/issues/17483

<img width="204" height="271" alt="grafik" src="https://github.com/user-attachments/assets/95f180a0-34ec-4d59-bd89-477d4d5fe754" />

<img width="198" height="247" alt="grafik" src="https://github.com/user-attachments/assets/2155bff6-e7e6-46b5-8ebb-8ab5036fafbc" />

<img width="179" height="177" alt="grafik" src="https://github.com/user-attachments/assets/32ed59db-4caa-4a0a-89c9-65ca7caef171" />

<img width="184" height="287" alt="grafik" src="https://github.com/user-attachments/assets/759bd775-fa43-4371-a3db-ef668951aa6d" />

PS: the `v(version)` in the screenshots are outdated 😅

Sysix commented on editors/vscode/client/extension.ts:
```diff
@@ -77,42 +77,25 @@ export async function activate(context: ExtensionContext) {
+    // Every tool is active
+  } else if (binaryPaths.every((path) => path)) {
+    statusBarItemHandler.setIcon("check-all");
+    // Some tools are inactive
```
This is the neat part! We do not care about the tools configuration

Sysix commented on editors/vscode/client/StatusBarItemHandler.ts:
```diff
@@ -19,23 +28,44 @@ export default class StatusBarItemHandler {
+      section.isEnabled = isEnabled;
+      section.content = text;
+      section.version = version ?? "unknown";
+      this.updateFullTooltip();
```
It is safe with type check

camc314 commented:
### Merge activity

* **Dec 31, 11:02 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 31, 11:02 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

Sysix committed:
refactor(vscode): improve statusbar item display (#17511)

This PR refactors the VSCode extension's status bar item display logic .
- Double Checkmark: Both tools (oxlint / oxfmt) are found and can be used
- Single Checkmark: One of the tools binary could not be found and is deactivated
- Circle: None of tools binary could be found

The background color changed a bit too. The error background is not shown anymore, only the warning background when no binaries could be found.

closes https://github.com/oxc-project/oxc/issues/17483

<img width="204" height="271" alt="grafik" src="https://github.com/user-attachments/assets/95f180a0-34ec-4d59-bd89-477d4d5fe754" />

<img width="198" height="247" alt="grafik" src="https://github.com/user-attachments/assets/2155bff6-e7e6-46b5-8ebb-8ab5036fafbc" />

<img width="179" height="177" alt="grafik" src="https://github.com/user-attachments/assets/32ed59db-4caa-4a0a-89c9-65ca7caef171" />

<img width="184" height="287" alt="grafik" src="https://github.com/user-attachments/assets/759bd775-fa43-4371-a3db-ef668951aa6d" />

PS: the `v(version)` in the screenshots are outdated 😅
